File: signl4-alert.js

// Send SIGNL4 alert from Node.js

// Your SIGNL4 team secret
const teamSecret = '<team-secret>'

const https = require('https')

// Alert data
const data = JSON.stringify({
    'Title': 'Alert',
    'Message': 'SIGNL4 alert from Node.js'
})

// SIGNL4 webhook URL
const options = {
  hostname: 'connect.signl4.com',
  port: 443,
  path: '/webhook/' + teamSecret,
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
}

const req = https.request(options, res => {
  console.log(`statusCode: ${res.statusCode}`)

  if (res.statusCode != 201) {
      // Error
      console.error('Error: ' + res.statusCode)
      return
  }

  // Success
  res.on('data', d => {
    process.stdout.write(d)
  })
})

// Error
req.on('error', error => {
  console.error(error)
})

req.write(data)
req.end()

File: signl4-alert.rs
// Send SIGNL4 alert from Rust

// In your Cargo.toml, add the following [dependencies]
// reqwest = { version = "0.11", features = ["json"] }
// serde_json = "1.0"
// tokio = { version = "1", features = ["full"] }

use reqwest::Error;
use serde_json::json;

#[tokio::main]
async fn main() -> Result<(), Error> {
    // Your SIGNL4 webhook URL
    let url = "https://connect.signl4.com/webhook/{team-secret}";

    // Create a JSON object
    let request = json!({
        "Title": "Alert",
        "Message": "SIGNL4 alert from Rust"
    });

    // Send the POST request with the JSON payload
    let client = reqwest::Client::new();
    let res = client.post(url)
        .json(&request)
        .send()
        .await?;

    // Print the response status and body
    println!("Response status: {}", res.status());
    let body = res.text().await?;
    println!("Response body: {}", body);

    Ok(())
}
